### PR TITLE
Send OnGazeEnter and OnGazeLeave messages when the user gazes on a hologram

### PR DESCRIPTION
**Pull request by dotMorten**
Addresses Issue #11

### CODE REVIEW

File: Assets/HoloToolkit/Input/GazeManager.cs [hunk 1]
```diff
@@ -62,6 +63,7 @@ private void UpdateRaycast()
                            MaxGazeDistance,
                            RaycastLayerMask);
 
+            GameObject oldHitObject = currentHitObject;
```
Comment (NeerajW) [inline Assets/HoloToolkit/Input/GazeManager.cs]: NIT: Please use similar naming convention as in GestureManager.cs.
oldHitObject to be oldFocusedObject
currentHitObject to focusedObject

File: Assets/HoloToolkit/Input/GazeManager.cs [hunk 2]
```diff
@@ -71,14 +73,23 @@ private void UpdateRaycast()
...
                 // If the raycast does not hit a hologram, default the position to last hit distance in front of the user,
                 // and the normal to face the user.
                 Position = gazeOrigin + (gazeDirection * lastHitDistance);
                 Normal = gazeDirection;
+                currentHitObject = null;
+            }
+            if(oldHitObject != currentHitObject) //The currently hit object has changed
+            {
+                if (oldHitObject != null)
+                    oldHitObject.SendMessage("OnGazeLeave");
```
Comment (NeerajW) [inline Assets/HoloToolkit/Input/GazeManager.cs]: Please follow existing brace pattern:

if(condition)
{
 // write code
}

File: Assets/HoloToolkit/Input/GazeManager.cs [hunk 3]
```diff
@@ -71,14 +73,23 @@ private void UpdateRaycast()
...
+                if (currentHitObject != null)
+                    currentHitObject.SendMessage("OnGazeEnter");
```
Comment (NeerajW) [inline Assets/HoloToolkit/Input/GazeManager.cs]: Please follow existing brace pattern:

if(condition)
{
 // write code
}

**Comment by dotMorten** [comment]
@NeerajW Thanks for the feedback. Updated!

**Pull request merged**